docs/getting-started/utility-apis.md

```diff
@@ -0,0 +1,140 @@
+  };
+
+  // the rest of the component ...
+};
+```
+
+Note that there is not explicit type given for the `ErrorApi`, this is because the `errorApiRef` has the type embedded, and `useApi` is able to infer the type.
+
+Also note that consuming Utility APIs is not limited to plugins, it can be done from any component inside Backstage, including the ones in `@backstage/core`. The only requirement is that they are beneath the `AppProvider` in the react tree.
+
+## Registering Utility API Implementations
+
+The Backstage App is responsible for providing implementations for all Utility APIs required by plugins. The example app in this repo registers its APIs inside [src/apis.ts](/packages/app/src/apis.ts). Here's an example of how to wire up the `ErrorApi` inside an app:
+
+```ts
+import {
+  ApiRegistry,
+  createApp,
+  alertApiRef,
+  errorApiRef,
+  AlertApiForwarder,
+  ErrorApiForwarder,
+} from '@backstage/core';
+
+const builder = ApiRegistry.builder();
+
+// The alert API is a self-contained implementation that shows alerts to the user.
+builder.add(alertApiRef, new AlertApiForwarder());
+
+// The error API uses the alert API to send error notifications to the user.
+builder.add(errorApiRef, new ErrorApiForwarder(alertApiForwarder));
+
+const app = createApp({
+  apis: apiBuilder.build(),
+  // ... other config
+});
+```
+
+The `ApiRegistry` is used to register all Utility APIs in the app and associate them with `ApiRef`s. It implements the `ApiHolder` interface, which enables it to provide an API implementation given an `ApiRef`.
+
+Note that our `ErrorApi` implementation depends on another Utility API, the `AlertApi`. This is the method with which APIs can depend on other APIs, using manual dependency injection at the initialization of the app. In general, if you want
+to depend on another Utility API in an implementation of an API, you import the type for that API and make it a
+constructor parameter.
+
+## Custom implementations of Utility APIs
+
+Defining a custom implementation of a utility API is easy, you simply need to export a class
+that `implements` the target API, for example:
+
+```ts
+export class IgnoringErrorApi implements ErrorApi {
+  post(error: Error, context?: ErrorContext) {
+    // ignore error
+  }
+}
+```
+
+The `IgnoringErrorApi` would then be imported in the app, and wired up like this:
+
+```ts
+builder.add(errorApiRef, new IgnoringErrorApi());
+```
+
+Note that the above line will cause an error if `IgnoreErrorApi` does not fully
+implement the `ErrorApi`, as it is checked by the type embedded in the `errorApiRef` at compile time.
+
+## Defining custom Utility APIs
+
+The pattern for plugins defining their own Utility APIs is not fully established yet. The current way is for the plugin to export it's own `ApiRef` and type for the API, along with one or more implementations. It is then up to the app to import, and register those APIs. See for example the lighthouse or graphiql plugins for examples of this.
```

Comment:
`export its own`

maybe make the words "lighthous" and "graphiql" link to the plugins, or even to where they export these apis